perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/MembersManagerBlImpl.java

```diff
@@ -2493,53 +2493,78 @@ private boolean isMemberInGracePeriod(Map<String, String> membershipExpirationRu
 	}
 
 	@Override
-	public void sendPasswordResetLinkEmail(PerunSession sess, Member member, String namespace, String url,
+	public void sendUsernameReminderEmail(PerunSession sess, Member member, final String namespace, String mailAddress,
+										  String language) {
+
+		User user = perunBl.getUsersManagerBl().getUserByMember(sess, member);
+
+		List<Attribute> logins = perunBl.getAttributesManagerBl().getLogins(sess, user);
+		boolean found = false;
+		if (logins != null) {
+			found = logins.stream().anyMatch(attr -> namespace.equals(attr.getFriendlyNameParameter()));
+		}
+		if (!found) {
+			throw new InternalErrorException(user.toString() + " doesn't have login in namespace: " + namespace);
+		}
```

Comment:
Couldn't we call `getUserLogin()` here, check for null, and save a few lines of code (and avoid more repetition in all of the email methods)?

Reply:
Have not noticed, while rewriting I have noticed the login has been extracted this way anyway. So this whole code block is not really needed.